File: practise/javascript/Anagram.js





function checkAnagram(str1,str2)
{
    if(str1.length!=str2.length)
        {
            return false
        }

       let da={};

       for( ab of str1)
        {
            da[ab]= (da[ab] || 0) + 1
        }
        console.log(da)

        for(let ab of str2)
            {
                if (da[ab]){
                    da[ab]=da[ab]-1
                }   
                else 
                {
                    da[ab]= (da[ab] || 0) + 1
                }
            }

            for(let ab in da)
                {
                    if(da[ab]>0)
                        {
                            return false
                        }
                }
       
console.log(da)
                return true;
}




console.log(checkAnagram("listen","silent"))

File: practise/Anagram.go
package main

import "fmt"

func main() {

	fmt.Println(checkAnagram("silent", "listen"))

}

func checkAnagram(str1 string, str2 string) bool {

	var checkAng = true
	if len(str1) != len(str2) {
		checkAng = false
	}

	var ab = make(map[string]int)

	for _, val := range str1 {
		ab[string(val)] = ab[string(val)] + 1
	}

	for _, val := range str2 {
		ab[string(val)]--
	}

	for _, val := range str2 {
		if ab[string(val)] != 0 {
			checkAng = false
			break
		}
	}

	// fmt.Println(ab)
	return checkAng
}
